int32 drill_req
int32 drill_height_value
int32 drill_width_value
